### PR TITLE
restructure

### PR DESCRIPTION
**Pull request by alienzj**
restructure

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,12 @@ cmake_minimum_required(VERSION 3.12)
...
 
-add_executable(metalog main.cc)
\ No newline at end of file
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}/bin)
+
+include_directories(include)
+
+add_executable(metalog example/metalog.cc)
+
+install(DIRECTORY include/ DESTINATION include)
```
Comment (Sleeplessy) [inline CMakeLists.txt:12]: I don't think wrapping the headers into an "include" dir for a head-only library is necessary.

maybe use  ` ${CMAKE_SOURCE_DIR}/metalog` is a better choice

**Review by Sleeplessy — CHANGES_REQUESTED** [review]
Thanks for your pull request. I've left some reviews opinions

**Review by Sleeplessy — APPROVED** [review]

**Review by Sleeplessy — DISMISSED** [review]
dismissed

File: CMakeLists.txt (hunk 1 above)
Comment (alienzj) [reply CMakeLists.txt:12]: https://github.com/avplayer/avhttp/blob/master/CMakeLists.txt#L78

**Pull request merged**